K2Bridge/Visitors/RangeVisitor.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Visitors
+{
+    using System.Globalization;
+    using K2Bridge.Models.Request.Aggregations;
+    using K2Bridge.Models.Response;
+
+    /// <content>
+    /// A visitor for the <see cref="RangeAggregation"/> element.
+    /// </content>
+    internal partial class ElasticSearchDSLVisitor : IVisitor
+    {
+        /// <inheritdoc/>
+        public void Visit(RangeAggregation rangeAggregation)
+        {
+            Ensure.IsNotNull(rangeAggregation, nameof(TermsAggregation));
+            EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Metric, nameof(RangeAggregation.Metric));
+            EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Field, nameof(RangeAggregation.Field));
+
+            // KustoQL case expression
+            // case(
+            var caseExpression = $"{KustoQLOperators.Case}(";
+
+            // case() predicates
+            // foo > 0 and foo < 800, "0-800"
+            foreach (var range in rangeAggregation.Ranges)
+            {
+                var rangeClause = new Models.Request.Queries.RangeClause()
+                {
+                    FieldName = rangeAggregation.Field,
+                    GTEValue = range.From?.ToString() ?? "*",
```

Comment:
in case we have to and from empty
we get
*-* 
?

if so maybe we can skip and not add this to the query (optimization)

Reply:
Actually this is a valid case, minus infinity to plus infinity, but this was not properly handled in the Kusto query... It is fixed (+ corresponding test).